js/src/UXCapture.js

```diff
@@ -2,44 +2,51 @@ import ExpectedMark from "./ExpectedMark";
 import View from "./View";
 
 const NOOP = () => {};
+const INTERACTION_START_MARK = "interactionStart";
+const NAVIGATION_START_MARK = "navigationStart";
 
 export default class UXCapture {
   constructor() {
     this.onMark = NOOP;
     this.onMeasure = NOOP;
+    this.startMark = NAVIGATION_START_MARK;
 
     window.UX = this;
   }
 
   /**
-   * Sets expected marks and corresponding zones for current view
+   * Creates a new View instance and supplies expected marks and
+   * corresponding zones to the View.
    *
-   * @TODO re-evaluate if we should allow multiple executions of this method
+   * TODO: re-evaluate if we should allow multiple executions of this method
+   *
+   * @param {object} zoneConfigs
    */
   expect(zoneConfigs) {
```

Comment:
I'll float this idea that I had mentioned to Sergey as well - I think we can avoid forcing the client to think about whether something is a _Page View_ or an _Interactive View_ by changing the `expect` interface to just be a `initView` method - regardless of whether it's the first view that appears or the 10th, the client just calls `initView` and the library determines what `startMark` to use - I don't think we need a `startTransition` method at all, since `initView` can handle updating the `startMark` reference, and we shouldn't expect the client to call `UX.configure` more than once.

From the client's perspective, the flow is:

```js
// Client app initializes UXCapture instance
window.UX = new UXCapture(config);
// or, UX.config(config) can be called lazily, although that seems unnecessary

// Client renders first view (Page View)
window.UX.initView(zoneConfigs);
// (expected marks fire)

// Client renders next view (Interactive View)
window.UX.initView(zoneConfigs)
// (expected marks fire)

// Client renders subsequent views...
window.UX.initView(zoneConfigs)
window.UX.initView(zoneConfigs)
window.UX.initView(zoneConfigs)
window.UX.initView(zoneConfigs)
...
```

This interface allows the client to be 'stateless' - only the UXCapture instance itself has to manage the state related to page view vs interactive view, which is great because that makes it much more difficult for the client to make mistakes.

The `initView` method would also be the _only_ method that manages `View` instances, which gives us very specific control over the view lifecycle - if we need a `view.destroy` method to clean up previous views before creating a new one, that can be done there.

The current PR implementation probably needs some more docs to make it clear what the client's responsibility is and what the complete lifecycle looks like. I might not be accounting for some edge cases, but I think we can (and should) simplify it here.

Reply:
Ah, yes, @sergeychernyshev has already set up a ticket related to this proposal: https://meetup.atlassian.net/browse/WP-829 👍

Reply:
I really like this, but, if we're going to do this, I highly suggest doing this now/first prior to any of the SPA tickets.

I'll move this conversation to slack, but maybe then I will put a pin in this branch, and quickly jump over to https://meetup.atlassian.net/browse/WP-829 and then come back to this.

cc @sergeychernyshev @mmcgahan